src/sentry/static/sentry/app/views/organizationIntegrations/installedPlugin.tsx

```diff
@@ -46,18 +48,26 @@ export class InstalledPlugin extends React.Component<Props> {
     );
   }
 
-  pluginUpdate = async (data: object) => {
+  pluginUpdate = async (data: object, method: 'POST' | 'DELETE' = 'POST') => {
     const {organization, projectItem, plugin} = this.props;
     // no try/catch so the caller will have to have it
     await this.props.api.requestPromise(
       `/projects/${organization.slug}/${projectItem.projectSlug}/plugins/${plugin.id}/`,
       {
-        method: 'POST',
+        method,
         data,
       }
     );
   };
 
+  updatePluginEnableStatus = async (enabled: boolean) => {
+    if (enabled) {
+      await this.pluginUpdate({enabled});
+    } else {
+      await this.pluginUpdate({enabled}, 'DELETE');
```

Comment:
@NisanthanNanthakumar I think you can make data `undefined` here

Reply:
@scefali it needs to be an object type. Maybe an empty object?

Reply:
@NisanthanNanthakumar that's fine